wen-restart/src/wen_restart.rs

```diff
@@ -405,6 +415,151 @@ fn find_bankhash_of_heaviest_fork(
     Ok(parent_bank.hash())
 }
 
+// Aggregate the heaviest fork and send updates to the cluster.
+pub(crate) fn aggregate_restart_heaviest_fork(
+    wen_restart_path: &PathBuf,
+    wait_for_supermajority_threshold_percent: u64,
+    cluster_info: Arc<ClusterInfo>,
+    bank_forks: Arc<RwLock<BankForks>>,
+    exit: Arc<AtomicBool>,
+    progress: &mut WenRestartProgress,
+) -> Result<()> {
+    let root_bank = bank_forks.read().unwrap().root_bank();
+    let epoch_stakes = root_bank.epoch_stakes(root_bank.epoch()).unwrap();
+    let total_stake = epoch_stakes.total_stake();
+    if progress.my_heaviest_fork.is_none() {
+        return Err(WenRestartError::UnexpectedState(RestartState::HeaviestFork).into());
+    }
+    let my_heaviest_fork = progress.my_heaviest_fork.clone().unwrap();
+    let heaviest_fork_slot = my_heaviest_fork.slot;
+    let heaviest_fork_hash = Hash::from_str(&my_heaviest_fork.bankhash)?;
+    let mut heaviest_fork_aggregate = HeaviestForkAggregate::new(
+        wait_for_supermajority_threshold_percent,
+        cluster_info.my_shred_version(),
+        epoch_stakes,
+        heaviest_fork_slot,
+        heaviest_fork_hash,
+        &cluster_info.id(),
+    );
+    if let Some(aggregate_record) = &progress.heaviest_fork_aggregate {
+        for (key_string, message) in &aggregate_record.received {
+            if let Err(e) = heaviest_fork_aggregate.aggregate_from_record(key_string, message) {
+                error!("Failed to aggregate from record: {:?}", e);
+            }
+        }
+    } else {
+        progress.heaviest_fork_aggregate = Some(HeaviestForkAggregateRecord {
+            received: HashMap::new(),
+            final_result: None,
+        });
+    }
+
+    let mut total_active_stake = heaviest_fork_aggregate.total_active_stake();
+    progress
+        .my_heaviest_fork
+        .as_mut()
+        .unwrap()
+        .total_active_stake = total_active_stake;
+    cluster_info.push_restart_heaviest_fork(
+        heaviest_fork_slot,
+        heaviest_fork_hash,
+        total_active_stake,
+    );
```

Comment:
We shouldn't be overloading gossip with these updates this often, I think the proposed  solution: https://github.com/anza-xyz/agave/pull/699/files#r1564923829 would avoid this

Reply:
As discussed, reusing RestartHeaviestFork for both cases. Also allowed 5% to not agree with us upon completion and added more logging info on failures.